pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -509,71 +510,83 @@ public static void addColumnMetadataInfo(PropertiesConfiguration properties, Str
+      properties.setProperty(getKeyFor(column, PARTITION_VALUES), columnIndexCreationInfo.getPartitions());
     }
 
     // datetime field
     if (fieldSpec.getFieldType().equals(FieldType.DATE_TIME)) {
       DateTimeFieldSpec dateTimeFieldSpec = (DateTimeFieldSpec) fieldSpec;
-      properties.setProperty(V1Constants.MetadataKeys.Column.getKeyFor(column, DATETIME_FORMAT),
-          dateTimeFieldSpec.getFormat());
-      properties.setProperty(V1Constants.MetadataKeys.Column.getKeyFor(column, DATETIME_GRANULARITY),
-          dateTimeFieldSpec.getGranularity());
+      properties.setProperty(getKeyFor(column, DATETIME_FORMAT), dateTimeFieldSpec.getFormat());
+      properties.setProperty(getKeyFor(column, DATETIME_GRANULARITY), dateTimeFieldSpec.getGranularity());
     }
 
-    Object defaultNullValue = columnIndexCreationInfo.getDefaultNullValue();
-    if (defaultNullValue instanceof byte[]) {
-      String defaultNullValueString = BytesUtils.toHexString((byte[]) defaultNullValue);
-      properties
-          .setProperty(V1Constants.MetadataKeys.Column.getKeyFor(column, DEFAULT_NULL_VALUE), defaultNullValueString);
+    String minValue = columnIndexCreationInfo.getMin().toString();
+    String maxValue = columnIndexCreationInfo.getMax().toString();
+    String defaultNullValue = columnIndexCreationInfo.getDefaultNullValue().toString();
+    if (dataType == DataType.STRING) {
+      // Check special characters for STRING column
+      if (isValidPropertyValue(minValue)) {
+        properties.setProperty(getKeyFor(column, MIN_VALUE), minValue);
+      }
+      if (isValidPropertyValue(maxValue)) {
+        properties.setProperty(getKeyFor(column, MAX_VALUE), maxValue);
+      }
+      if (isValidPropertyValue(defaultNullValue)) {
+        properties.setProperty(getKeyFor(column, DEFAULT_NULL_VALUE), defaultNullValue);
+      }
     } else {
-      properties.setProperty(V1Constants.MetadataKeys.Column.getKeyFor(column, DEFAULT_NULL_VALUE),
-          String.valueOf(defaultNullValue));
+      properties.setProperty(getKeyFor(column, MIN_VALUE), minValue);
+      properties.setProperty(getKeyFor(column, MAX_VALUE), maxValue);
+      properties.setProperty(getKeyFor(column, DEFAULT_NULL_VALUE), defaultNullValue);
     }
   }
 
   public static void addColumnMinMaxValueInfo(PropertiesConfiguration properties, String column, String minValue,
       String maxValue) {
-    properties.setProperty(getKeyFor(column, MIN_VALUE), minValue);
-    properties.setProperty(getKeyFor(column, MAX_VALUE), maxValue);
+    // Check special characters for STRING column
+    if (isValidPropertyValue(minValue)) {
+      properties.setProperty(getKeyFor(column, MIN_VALUE), minValue);
+    }
+    if (isValidPropertyValue(maxValue)) {
+      properties.setProperty(getKeyFor(column, MAX_VALUE), maxValue);
+    }
+  }
+
+  /**
+   * Helper method to check whether the given value is a valid property value.
+   * <p>Value is invalid iff:
+   * <ul>
```

Comment:
Not sure how we find out that these are the only comprehensive rules

Reply:
In `SegmentColumnarIndexCreatorTest.testPropertyValueWithSpecialCharacters()`, generating random strings to verify that the same string can be retrieved